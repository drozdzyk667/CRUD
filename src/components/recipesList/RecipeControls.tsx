import React from 'react';
import Icon from 'Icons';
import { ButtonsContainer, Button } from '../styled';
import { makeStyles, Box } from '@material-ui/core';
import {
  Recipe,
  Names,
  Colors,
  WARNING_MESSAGE
} from 'pages/recipes/Recipes.constants';
import ConfirmationModal from '../confirmation';

const useStyles = makeStyles({
  container: {
    display: 'flex',
    flexDirection: 'row',
    alignItems: 'center'
  },
  text: {
    margin: '0 0.5em'
  }
});

interface RecipeControlProps {
  recipe: Recipe;
  handleOpenForm: (type: string, element?: Recipe) => void;
  handleDeleteRecipe: (id: number) => void;
}

const RecipeControls: React.FC<RecipeControlProps> = ({
  recipe,
  handleOpenForm,
  handleDeleteRecipe
}) => {
  const classes = useStyles();
  const [isDeleteModal, setIsDeleteModal] = React.useState(false);

  const toggleConfirmationModal = () => {
    setIsDeleteModal(isDeleteModal => !isDeleteModal);
  };

  const handleConfirmDelete = () => {
    handleDeleteRecipe(recipe.id);
    setIsDeleteModal(false);
  };

  return (
    <>
      <ButtonsContainer>
        <Button background={Colors.delete} onClick={toggleConfirmationModal}>
          <Box className={classes.container}>
            <span className={classes.text}>{Names.delete}</span>
            <Icon name='Delete' />
          </Box>
        </Button>
        <Button
          black
          background={Colors.edit}
          onClick={() => handleOpenForm(Names.edit, recipe)}
        >
          <Box className={classes.container}>
            <span className={classes.text}>{Names.edit}</span>
            <Icon name='Edit' />
          </Box>
        </Button>
      </ButtonsContainer>
      <ConfirmationModal
        isDeleteModal={isDeleteModal}
        handleConfirmDelete={handleConfirmDelete}
        toggleConfirmationModal={toggleConfirmationModal}
        message={WARNING_MESSAGE}
      />
    </>
  );
};

export default RecipeControls;
