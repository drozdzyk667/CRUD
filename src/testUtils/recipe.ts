export const RecipeMock = {
  id: 1,
  recipeName: 'Test Recipe',
  isExpanded: false,
  ingredients: ['test1', 'test2', 'test3']
};
