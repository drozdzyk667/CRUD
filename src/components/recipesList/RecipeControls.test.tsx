import React from 'react';
import { shallow } from 'enzyme';
import RecipeControls from './RecipeControls';
import { RecipeMock } from 'testUtils/recipe';

describe('RecipeElement', () => {
  const clickFnOpenForm = jest.fn();
  const clickFnDeleteRecipe = jest.fn();

  it('should render correctly ', () => {
    const component = shallow(
      <RecipeControls
        handleOpenForm={clickFnOpenForm}
        handleDeleteRecipe={clickFnDeleteRecipe}
        recipe={RecipeMock}
      />
    );
    expect(component).toMatchSnapshot();
  });

  it('should render Delete', () => {
    const component = shallow(
      <RecipeControls
        handleOpenForm={clickFnOpenForm}
        handleDeleteRecipe={clickFnDeleteRecipe}
        recipe={RecipeMock}
      />
    );
    const btn = component.find({
      'data-testid': 'Delete'
    });
    expect(btn.first().props().children).toEqual('Delete');
  });

  it('should render Edit', () => {
    const component = shallow(
      <RecipeControls
        handleOpenForm={clickFnOpenForm}
        handleDeleteRecipe={clickFnDeleteRecipe}
        recipe={RecipeMock}
      />
    );
    const btn = component.find({
      'data-testid': 'Edit'
    });
    expect(btn.first().props().children).toEqual('Edit');
  });
  it('should render delete button', () => {
    const component = shallow(
      <RecipeControls
        handleOpenForm={clickFnOpenForm}
        handleDeleteRecipe={clickFnDeleteRecipe}
        recipe={RecipeMock}
      />
    );
    const btn = component.find({
      'data-testid': 'delete-button'
    });
    expect(btn.first().length).toBe(1);
  });
  it('should render edit button', () => {
    const component = shallow(
      <RecipeControls
        handleOpenForm={clickFnOpenForm}
        handleDeleteRecipe={clickFnDeleteRecipe}
        recipe={RecipeMock}
      />
    );
    const btn = component.find({
      'data-testid': 'edit-button'
    });
    expect(btn.first().length).toBe(1);
  });
});
