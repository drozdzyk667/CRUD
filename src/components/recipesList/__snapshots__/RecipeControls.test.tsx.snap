// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RecipeElement should render correctly  1`] = `
<Fragment>
  <styled.div>
    <styled.button
      background="#ed5b3e"
      data-testid="delete-button"
      onClick={[Function]}
    >
      <Styled(MuiBox)
        className="makeStyles-container-1"
      >
        <span
          className="makeStyles-text-2"
          data-testid="Delete"
        >
          Delete
        </span>
        <ForwardRef
          name="Delete"
        />
      </Styled(MuiBox)>
    </styled.button>
    <styled.button
      background="#FFF"
      black={true}
      onClick={[Function]}
    >
      <Styled(MuiBox)
        className="makeStyles-container-1"
        data-testid="edit-button"
      >
        <span
          className="makeStyles-text-2"
          data-testid="Edit"
        >
          Edit
        </span>
        <ForwardRef
          name="Edit"
        />
      </Styled(MuiBox)>
    </styled.button>
  </styled.div>
  <ConfirmationModal
    handleConfirmDelete={[Function]}
    isConfirmationModalOpen={false}
    message="Are you sure?, Recipe will be lost !"
    toggleConfirmationModal={[Function]}
  />
</Fragment>
`;
